pkg/rpc/context.go

```diff
@@ -2204,18 +2264,90 @@ func (rpcCtx *Context) grpcDialNodeInternal(
 ) *Connection {
 	k := connKey{target, remoteNodeID, class}
 	if conn, ok := rpcCtx.m.Get(k); ok {
-		// There's a cached connection.
+		// Returns cached connection that isn't neither failed nor decommissioned.
 		return conn
 	}
-
 	ctx := rpcCtx.makeDialCtx(target, remoteNodeID, class)
 
-	conn, inserted := rpcCtx.m.TryInsert(k)
+	conn, pc, inserted := rpcCtx.m.TryInsert(k)
 	if !inserted {
 		// Someone else won the race.
 		return conn
 	}
 
+	// TODO (koorosh): breaker is stored per `peer` and shouldn't be reinitialized every time `grpcDialNodeInternal`
+	// is called.
+	if pc.breaker == nil {
+		// TODO (koorosh): Probably this logic should be extracted from here...
```

Comment:
+1 maybe to `rpc/breaker.go` because you'll ultimately have to disable the existing breaker functionality I think! That functionality mostly powers `(*nodedialer.Dialer).getBreaker` but now components that sit on top of `rpc.Context` don't need their own breaker any more, they'll just blindly dial and the `rpc.Context` will fail-fast as appropriate.